src/main/java/eu/dissco/backend/repository/DigitalMediaObjectRepository.java

```diff
@@ -25,55 +26,50 @@ public class DigitalMediaObjectRepository {
+    return context.select(DIGITAL_MEDIA_OBJECT.asterisk())
+        .from(DIGITAL_MEDIA_OBJECT)
         .offset(offset).limit(pageSizePlusOne).fetch(this::mapToMultiMediaObject);
   }
 
-  public DigitalMediaObject getLatestDigitalMediaObjectById(String id) {
-    return context.select(NEW_DIGITAL_SPECIMEN.SPECIMEN_NAME, NEW_DIGITAL_SPECIMEN.VERSION,
-            NEW_DIGITAL_SPECIMEN.ID, NEW_DIGITAL_MEDIA_OBJECT.asterisk())
-        .from(NEW_DIGITAL_SPECIMEN)
-        .join(NEW_DIGITAL_MEDIA_OBJECT)
-        .on(NEW_DIGITAL_SPECIMEN.ID.eq(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID))
-        .where(NEW_DIGITAL_MEDIA_OBJECT.ID.eq(id))
+  public DigitalMediaObjectWrapper getLatestDigitalMediaObjectById(String id) {
+    return context.select(DIGITAL_MEDIA_OBJECT.asterisk())
+        .from(DIGITAL_MEDIA_OBJECT)
+        .where(DIGITAL_MEDIA_OBJECT.ID.eq(id))
         .fetchOne(this::mapToMultiMediaObject);
   }
 
-  public List<DigitalMediaObject> getDigitalMediaForSpecimen(String id) {
-    return context.select(NEW_DIGITAL_MEDIA_OBJECT.asterisk())
-        .from(NEW_DIGITAL_MEDIA_OBJECT)
-        .where(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(id))
+  public List<DigitalMediaObjectWrapper> getDigitalMediaForSpecimen(String id) {
+    return context.select(DIGITAL_MEDIA_OBJECT.asterisk())
+        .from(DIGITAL_MEDIA_OBJECT)
+        .where(DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(id))
         .fetch(this::mapToMultiMediaObject);
   }
 
   public List<String> getDigitalMediaIdsForSpecimen(String id) {
-    return context.select(NEW_DIGITAL_MEDIA_OBJECT.ID)
-        .from(NEW_DIGITAL_MEDIA_OBJECT)
-        .where(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(id))
+    return context.select(DIGITAL_MEDIA_OBJECT.ID)
+        .from(DIGITAL_MEDIA_OBJECT)
+        .where(DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(id))
         .fetch(Record1::value1);
   }
 
-  private DigitalMediaObject mapToMultiMediaObject(Record dbRecord) {
+  private DigitalMediaObjectWrapper mapToMultiMediaObject(Record dbRecord) {
     try {
-      return new DigitalMediaObject(
-          HANDLE_STRING + dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ID),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.VERSION),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.CREATED),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.TYPE),
-          HANDLE_STRING + dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.MEDIA_URL),
-          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.FORMAT),
-          HANDLE_STRING + dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.SOURCE_SYSTEM_ID),
-          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DATA).data()),
-          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ORIGINAL_DATA).data()));
+      var digitalMediaObject = mapper.readValue(dbRecord.get(DIGITAL_MEDIA_OBJECT.DATA).data(),
+              DigitalEntity.class)
```

Comment:
It doesn't enforce the required fields. So this doesn't fail. I do the readValue here so that we won't have to map the whole object only add the fields that are missing.